### ISSUE TITLE
QuantifiedConstraints aren't printed correctly:

### ISSUE THREAD

**Issue by phadej** [haskell/haddock#877]
![screenshot from 2018-07-09 10-53-13](https://user-images.githubusercontent.com/51087/42437657-6884337a-8366-11e8-9662-d92740aadaf6.png)

The correct is:

```haskell
 (forall f. Functor f => Functor (t f)) => Bound (ScopeT n t)
```

Note: If I add an  extra constraint, so the context is a tuple, parens are printed, so the output is correct

![screenshot from 2018-07-09 10-55-32](https://user-images.githubusercontent.com/51087/42437749-b2f2cec6-8366-11e8-8110-13f5a0e73ff2.png)

EDIT: tried with `ghc-8.6.0.20180627`

**Comment by harpocrates** [comment]
Fix is on its way in #874. For example,

```haskell
{-# LANGUAGE QuantifiedConstraints #-}
module QuantifiedConstraints where

class Foo a where
  fooed :: a

needsParensAroundContext :: (forall x. Foo (f x)) => f Int
needsParensAroundContext = fooed

needsNoParensAroundContext :: Foo (f Int) => f Int
needsNoParensAroundContext = fooed
```

now turns into this

<img width="574" alt="screen shot 2018-07-09 at 6 55 11 am" src="https://user-images.githubusercontent.com/10766081/42454716-1355422a-8345-11e8-8e48-0b241dbdb6f4.png">

(The fix also works for class contexts in instance lists.)

**Comment by phadej** [comment]
@harpocrates cool!

**Comment by harpocrates** [comment]
Fixed in #874.
